spark/src/main/scala/io/substrait/spark/expression/ToSubstraitExpression.scala

```diff
@@ -196,7 +199,10 @@ abstract class ToSubstraitExpression extends HasOutputStack[Seq[Attribute]] {
       case InSet(value, set) => translateIn(value, set.toSeq.map(v => Literal(v)))
       case scalar @ ScalarFunction(children) =>
         Util
-          .seqToOption(children.map(translateUp))
+          .seqToOption(children.map {
+            case Enum(value) => Some(ImmutableEnumArg.builder.value(Optional.of(value)).build)
```

Comment:
nit: if EnumArg doesn't have the builder on itself, you could add it there so that you could do just `EnumArg.builder...`